app.js

```diff
@@ -0,0 +1,330 @@
+let childScreen = document.getElementById("child-screen");
+let mainScreen = document.getElementById("output-screen");
+let errorMessage = document.getElementById("error-message");
+let operators = ["%", "+", "-", "*", "/", ".", "^", ".e+0"];
+let memoryItems = [];
+checkMemory();
+
+// Display user input on screen
+function displayEntry(value) {
+  if (mainScreen.innerHTML == "0" && value != ".") {
+    if (operators.includes(value)) {
+      mainScreen.innerHTML += value;
+    } else {
+      mainScreen.innerHTML = value;
+    }
+  } else if (mainScreen.innerHTML.substr(mainScreen.innerHTML.length - 4) == ".e+0") {
+    if (operators.includes(value)) {
+      mainScreen.innerHTML = mainScreen.innerHTML;
+    } else {
+      mainScreen.innerHTML = mainScreen.innerHTML.slice(0, -1) + value;
+    }
+  }  
+    else {
+    if (ValidateInput(value) && mainScreen.innerHTML != "") {
+      mainScreen.innerHTML += value;
+    } else {
+      mainScreen.innerHTML = mainScreen.innerHTML.slice(0, -1) + value;
+    }
+  }
+}
+
+function displayOutput(value){
+  if(isNaN(value) || !isFinite(value)){
+    showError("Invalid Input");
+    mainScreen.innerHTML = "0";
+  }else{
+    mainScreen.innerHTML = value;
+  }
+}
+
+// Calculate
+function Calculator() {
+  try {
+    if (mainScreen.innerHTML != "" && !operators.includes(mainScreen.innerHTML.substr(mainScreen.innerHTML.length - 1))) {
+      if (mainScreen.innerHTML.includes("^")) {
+        const findXYSqaure = mainScreen.innerHTML.split("^", 2);
+        childScreen.innerHTML = mainScreen.innerHTML;
+        displayOutput(Math.pow(findXYSqaure[0], findXYSqaure[1]));
+      } else {
+        childScreen.innerHTML = mainScreen.innerHTML;
+        let count = eval(mainScreen.innerHTML);
+        displayOutput(count);
+      }
+    }else{
+      showError("Please enter valid input");
+    }
+  } catch (err) {
+    showError("Please enter valid input");
```

Comment:
Can we assign this text message in variable and then use it?

Reply:
yes sir, I will change it.